pandas/core/apply.py

```diff
@@ -274,13 +274,16 @@ def apply_standard(self):
 
         # we cannot reduce using non-numpy dtypes,
         # as demonstrated in gh-12244
-        if (
+        flag = (
             self.result_type in ["reduce", None]
             and not self.dtypes.apply(is_extension_array_dtype).any()
             # Disallow complex_internals since libreduction shortcut
             #  cannot handle MultiIndex
             and not isinstance(self.agg_axis, ABCMultiIndex)
-        ):
+        )
```

Comment:
what happens if you change this condition to also rule out mixed-dtypes?  that would prevent the call to self.values which is what does the object-dtype casting that is causing the problem

Reply:
Done. I forgot to add this additional rule before